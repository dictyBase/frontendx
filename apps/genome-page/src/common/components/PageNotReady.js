// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesome from "react-fontawesome"
import { SaveButton, Jumbotron, RouterLink } from "styles/style"

/**
 * General 404 error page, currently designated as "Page Not Ready"/"Under Construction"
 */

const PageNotReady = () => {
  return (
    <Flex justify="center" style={{ marginTop: "20px" }}>
      <Box w="60%">
        <Jumbotron>
          <h1>
            <FontAwesome name="wrench" /> Under Construction
          </h1>
          <p>This page is not ready yet.</p>
          <Flex justify="center">
            <Box w={["90%", "60%", "25%"]}>
              <RouterLink to="/">
                <SaveButton size="small" variant="contained" color="primary">
                  Dictybase Home
                </SaveButton>
              </RouterLink>
            </Box>
          </Flex>
        </Jumbotron>
      </Box>
    </Flex>
  )
}

export default PageNotReady
