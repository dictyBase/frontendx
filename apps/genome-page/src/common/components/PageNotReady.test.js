import React from "react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import PageNotReady from "./PageNotReady"

test("matching a snapshot of PageNotReady", () => {
  const component = renderer.create(
    <BrowserRouter>
      <PageNotReady />
    </BrowserRouter>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
