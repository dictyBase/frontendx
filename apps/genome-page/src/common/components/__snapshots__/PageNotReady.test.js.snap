// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of PageNotReady 1`] = `
.c1 {
  color: #428bca;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  padding-top: 30px;
  padding-bottom: 30px;
  margin-bottom: 30px;
  color: inherit;
  background-color: #eee;
  text-align: center;
}

.c0 h1 {
  color: inherit;
}

.c0 p {
  margin-bottom: 15px;
  font-size: 21px;
  font-weight: 200;
}

.c2.c2 {
  width: 100%;
  background-color: #15317e;
}

<div
  className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
  style={
    Object {
      "marginTop": "20px",
    }
  }
>
  <div
    className="ilqAEZ"
    is={null}
  >
    <div
      className="c0"
    >
      <h1>
        <span
          aria-hidden={true}
          className="fa fa-wrench"
        />
         Under Construction
      </h1>
      <p>
        This page is not ready yet.
      </p>
      <div
        className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="eyMZla"
          is={null}
        >
          <a
            className="c1"
            href="/"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-contained-10 MuiButton-containedPrimary-11 MuiButton-raised-13 MuiButton-raisedPrimary-14 MuiButton-sizeSmall-22 c2"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-2"
              >
                Dictybase Home
              </span>
              <span
                className="MuiTouchRipple-root-28"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
